File: Question2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    let dummy = new ListNode();
    let pointer = dummy;
    let overflow = 0;
    while(l1 !== null || l2 !== null){
        v1 = l1 == null? 0:l1.val;
        v2 = l2 == null? 0:l2.val;
        value = (v1+v2+overflow)%10;
        overflow = Math.floor((v1+v2+overflow)/10);
        let node = new ListNode(value);
        pointer.next = node;
        pointer = pointer.next;
        if(l1 !== null){
            l1 = l1.next;
        }
          if(l2 !== null){
            l2 = l2.next;
        }
    }
    if(overflow !== 0) pointer.next = new ListNode(overflow);
    return dummy.next;
};

File: java/src/com/abbie/Question2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       int carry = 0;
       ListNode prev = new ListNode(0);
        ListNode head = prev;
    while (l1 != null || l2 != null || carry != 0){
            ListNode cur = new ListNode(0);
            int sum  = ((l1 == null)?0:l1.val) + ((l2 == null)?0:l2.val) + carry;
            cur.val = sum%10;
            carry = sum/10;
            prev.next = cur;
            prev = cur;
            
            l1 = ((l1 == null)?l1:l1.next);
            l2 = ((l2 == null)?l2:l2.next);
        }
        return head.next;
    }
}